accepted/00000-hub-reporting.md

```diff
@@ -0,0 +1,208 @@
+- setup database users,
+- setup tasktomatic jobs.
+
+All the other information required (e.g. Uyuni Server info) will be retrieved by already existing configuration.
+
+## Reporting Database
+A reporting DB is setup in all Uyuni Servers and Uyuni Hub nodes to store reporting data.
+The default setup will create a Read/Write user, to manage the database and write the data from it into the Reporting DB.
+This user and the DB connection parameters are written into `/etc/rhn/rhn.conf` similar to the default DB options.
+
+## Initial Setup
+Uyuni Hub should create its own read-only user on the Reporting Databases of the single Servers.
+As the Servers are managed with salt, we will write a state to create an account.
+The username and password are generated on the Hub and provided as pillar data.
+On the server the state take care of the existence of the account and the Hub
+can store the parameters in its database under the system entry.
+
+This account does not need to be stored in plain text on the managed Server and can be kept only on the Hub.
+The same apply for accounts created by the administrator used for Reporting Tools which need access
+to the Reporting Database.
+
+
+# Workflow
+
+## Scheduled job: collect from Uyuni DB
+A taskomatic job is responsible to fetch and prepare the data from the main Uyuni
+Database and insert them into the Reporting Database.
+The job should be written in a way, that the Reporting Database could be on different Host.
+
+Keeping the code which insert the data into the reporting DB in sync with the Reporting Database Schema
+is a requirement.
+
+Implementing a taskomatic simple java job should be sufficient as we need only one task which run at a
+certain point in time.
+
+We will create Views in the main Uyuni Database which prepare the data. The job should just query all and insert
+the data into the reporting DB. We will get schema errors in case the schema of the main Uyuni Database change.
+
+As we design this feature to be ready to use an external database for reporting, we do not use materialized views.
+
+## Scheduled job: collect remote data
+We also have an additional taskomatic job which collect the data from all the managed
+Uyuni servers and insert them into the Hub Reporting Database which could be
+again an external DB.
+We must parallelize the jobs to be able to gather the data from all Servers even in a large environment.
+The goal is to get the data from 1000 Servers in maximal 3 hours.
+
+The job get the list of candidates from the Hub Database including the user and connection paramaters to the
+reporting databases of the managed Uyuni Servers.
+
+The reporting DB does not use own ID columns with own sequences. To keep things simple we will
+remove all rows for the Uyuni Server identified by its `mgm_id` and insert all which we get new from the it.
+In case we see performance problems we need to re-think this approach and need to implement a more inteligent
```

Comment:
I'm not sure if it should be part of this RFC, but with this amount of delete/insert, statistics may need to be regenerated for the target tables.

Reply:
I tend to consider that an implementation detail, if and when we detect actual performance problems directly tied to statistics.